internal/uvm/create_lcow.go

```diff
@@ -298,11 +301,21 @@ func CreateLCOW(ctx context.Context, opts *OptionsLCOW) (_ *UtilityVM, err error
 				ImageFormat: imageFormat,
 			},
 		}
-		// Add to our internal structure
-		uvm.vpmemDevices[0] = &vpmemInfo{
-			hostPath: opts.RootFSFile,
-			uvmPath:  "/",
-			refCount: 1,
+		if uvm.vpmemMultiMapping {
+			pmem := newPackedVPMemDevice()
+			pmem.maxMappedDeviceCount = 1
+			memReg, err := pmem.Allocate(4 * memory.GigaByte)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to allocate memory for rootfs")
+			}
+			dev := newVPMemMappedDevice(opts.RootFSFile, "/", 4*memory.GigaByte, memReg)
+			if err := pmem.mapVHDLayer(ctx, dev); err != nil {
+				return nil, errors.Wrapf(err, "failed to save internal state for a multi-mapped rootfs device")
+			}
+			uvm.vpmemDevicesPacked[0] = pmem
+		} else {
+			dev := newDefaultVPMemInfo(opts.RootFSFile, "/")
+			uvm.vpmemDevicesDefault[0] = dev
```

Comment:
This is just for the rootfs vhd for the vm, do we actually gain anything from using device mapper for this? I might not be fully following, but it looks like we set the maximum allowable number of mapped devices to 1 in the dm case anyways.

Reply:
no we don't. I just allocate entire VPMem for rootfs. since we're setting the document for it manually here it doesn't seem to work properly with multi-mapped layers.